html-Detector/SubSystemExample/databaseQueries/index.html

```diff
@@ -0,0 +1,387 @@
+                import { dbTable } from '/includes/tooldaq.js';
+                const query = "SELECT * FROM monitoring LIMIT 10";
+                const headers = ["Time", "Device", "Data"]; // Example headers
+                function runDbHtmlTable() {
+                  dbHtmlTable(query, headers).then(data => {
+                    console.log("dbTable result:", data);
+                    const container = document.getElementById("tableContainer");
+                    container.innerHTML = data;
+                  });
+                }
+              </code>
+            </pre>
+            <p>
+              <strong>Output:</strong> The query results will be displayed in an HTML table within a container element
+              with
+              the ID <code>tableContainer</code>.
+            </p>
+            <div id="dbHtmlTableContainer" class="demo-container mdl-shadow--2dp">
+              <button onclick="runDbHtmlTableDemo()"
+                class="mdl-button mdl-js-button mdl-button--raised mdl-button--colored">
+                Run dbHtmlTable Demo
+              </button>
+              <div id="dbHtmlTableOutput" class="mdl-typography--body-1"></div>
+            </div>
+
+            <script>
+              function runDbHtmlTableDemo() {
+                import('/includes/tooldaq.js').then(({ dbHtmlTable }) => {
+                  const query = "SELECT * FROM monitoring LIMIT 10";
+                  const headers = ["Time", "Device", "Data"];
+
+                  dbHtmlTable(query, headers).then(data => {
+                    console.log("dbHTmlTable result:", data);
+                    const container = document.getElementById("dbHtmlTableContainer");
+                    container.innerHTML = data;
+                  }).catch(error => {
+                    document.getElementById('dbTableOutput').innerHTML = `<p style="color:red;">Error: ${error.message}</p>`;
+                  });
+                });
+              }
+            </script>
+            <hr>
+          </div>
+
+          <div id="db-plot-example">
+            <h3>dbPlot(query, template)</h3>
+            <p>
+              <strong>Description:</strong> Executes a SQL query and generates a plot from the query results.
+              This is ideal for visualisations and plots.
+            </p>
+            <p>
+              <strong>Parameters:</strong> <br>
+              <strong>query</strong> is a string with an SQL query. Parameters interpolation and string escaping are currently not supported. <br>
+                <strong>template</strong> can be used to set default values for all returned traces. It should be an object (possibly null)
+            </p>
+            <p><strong>Example:</strong></p>
+            <pre>
+              <code class="language-javascript">
+                import { dbPlot } from '/includes/tooldaq.js';
+
+                const query = "SELECT time, value FROM monitoring ORDER BY time LIMIT 50";
```

Comment:
This query is invalid: there is no `value` column in the `monitoring` table. And I got a wrong error message here because the error reporting code attempted to read the response body twice. Will submit a pull request shortly.

Also, at this point the user is likely to have an empty database --- except for `middleman` which produces data that is not interesting to plot. I suggest faking some data, e.g.,
```
select i, 100 * random() from generate_series(0, 9) i
```